### DISCUSSION TITLE
What happens when the storage at 'LOTUS_PATH' runs out?

### DISCUSSION THREAD

**Discussion by LoganC034** [filecoin-project/lotus#9341] [Build and Sync] [answered]
Still trying to get my head wrapped around managing the data store for a lotus node. I am mostly using it as a means to interact with the command line wallet without relying on a remote node. I would like to avoid committing too much space to it for this limited use case. I know there is a way to stop the node compact and re-sync every so often and suppose that this could be automated. But I was wondering what would happen if you simply let it run out of space. Does it crash? Does it clean up after itself? 

semi-related, would be nice to have a rolling snapshot that maintains the lightweight state of current block minus 2000. Rather than building on top of  lightweight store , it would  take a first in first out approach and deletes the oldest block with every new block added. I will admit, not a whole lot of thought has been put into this on my end.

**Comment by stuberman** [comment, marked as answer]
For the Lotus node, you never want to run out of disk space. One good solution is to use the split store feature for the chain and then separate the cold store [chain] and symlink it to a larger HDD. Keep the rest of the chain ([splitstore](https://github.com/filecoin-project/lotus/discussions/5788)) on fast disk like NVMe.

`du -sch .lotus/datastore/* | sort -rh`

> 1.2T	total
> 1019G	/.lotus/datastore/chain
> 163G	/.lotus/datastore/splitstore
> 51M	/.lotus/datastore/metadata
> 848K	/.lotus/datastore/client
> 20K	/.lotus/datastore/staging

**Reply by stuberman** [reply]
BTW -I use the discard option to keep the chain small.
There is a newer splitstore [update](https://github.com/filecoin-project/lotus/releases/tag/v1.17.1) being tested right now as well

**Reply by LoganC034** [reply]
Oh this update is intriguing. And thank you for your response.

**Reply by rjan90** [reply]
> But I was wondering what would happen if you simply let it run out of space. Does it crash? Does it clean up after itself?

Currently the node will crash and corrupt the datastore if you run out of space! We have a fairly new enhancement request https://github.com/filecoin-project/lotus/issues/9312 for lotus to verify the space left and to stop sync process if there is less than 1GB of free disk space available. I think the enhancement request makes sense and would be a good UX improvement!

The developers will probably need to have a look if its feasible to implement this without using too much extra resources though.

And as @stuberman noted  SplitStore is a really good option for keeping the chain small without having to prune the chain manually 😄
